examples/tv-app/android/java/AppImpl.h

```diff
@@ -87,17 +89,23 @@ class DLL_EXPORT ContentAppImpl : public ContentApp
 {
 public:
     ContentAppImpl(const char * szVendorName, uint16_t vendorId, const char * szApplicationName, uint16_t productId,
-                   const char * szApplicationVersion, const char * setupPIN, ContentAppAttributeDelegate * attributeDelegate) :
+                   const char * szApplicationVersion, const char * setupPIN, ContentAppAttributeDelegate * attributeDelegate,
+                   ContentAppCommandDelegate * commandDelegate) :
         mApplicationBasicDelegate(kCatalogVendorId, BuildAppId(vendorId), szVendorName, vendorId, szApplicationName, productId,
                                   szApplicationVersion),
-        mAccountLoginDelegate(setupPIN), mContentLauncherDelegate(attributeDelegate, { "image/*", "video/*" },
-                                                                  to_underlying(SupportedStreamingProtocol::kDash) |
-                                                                      to_underlying(SupportedStreamingProtocol::kHls)),
+        mAccountLoginDelegate(commandDelegate, setupPIN),
+        mContentLauncherDelegate(attributeDelegate, { "image/*", "video/*" },
+                                 to_underlying(SupportedStreamingProtocol::kDash) |
+                                     to_underlying(SupportedStreamingProtocol::kHls)),
         mMediaPlaybackDelegate(attributeDelegate),
         mTargetNavigatorDelegate(attributeDelegate, { "home", "search", "info", "guide", "menu" }, 0){};
     virtual ~ContentAppImpl() {}
 
-    AccountLoginDelegate * GetAccountLoginDelegate() override { return &mAccountLoginDelegate; };
+    AccountLoginDelegate * GetAccountLoginDelegate() override
+    {
+        mAccountLoginDelegate.SetEndpointId(GetEndpointId());
```

Comment:
nit: this is now a getter with side-effects (which is unsually unexpected of getters). Consider at some point renaming the method to make it more obvious that it is not what one would expect from a getter or at least add a comment describing the validity of the return value.

Reply:
Ack. I think there may be some other ways to refactor to avoid this. The issue currently is that the content app object is created before the endpoint is assigned. And once the endpoint is assigned the control does not come(yet) to the app to be able to do this well. There may be a way to override the SetEndpointID on the base class. Will take it up as a separate refactor.